### PR TITLE
Add GET/{manager,cluster/:node_id}/config/:component/:configuration endpoints

### PR DESCRIPTION
**Pull request by mgmacias95**
Hello team,

This PR adds API support for https://github.com/wazuh/wazuh/issues/2988.

Mocha tests:
```javascript
# mocha test/test_cluster.js --grep component


  Cluster
    GET/cluster/master/config/:component/:configuration
      ✓ Request-Agentless-Agentless (314ms)
      ✓ Request-Analysis-Global (307ms)
      ✓ Request-Analysis-Active-response (337ms)
      ✓ Request-Analysis-Alerts (294ms)
      ✓ Request-Analysis-Command (287ms)
      ✓ Request-Analysis-Internal (297ms)
      ✓ Request-Auth-Auth (307ms)
      ✓ Request-Com-Active-response (295ms)
      ✓ Request-Com-Internal (280ms)
      ✓ Request-Csyslog-Csyslog (297ms)
      ✓ Request-Integrator-Integration (291ms)
      ✓ Request-Logcollector-Localfile (303ms)
      ✓ Request-Logcollector-Socket (289ms)
      ✓ Request-Logcollector-Internal (302ms)
      ✓ Request-Mail-Global (282ms)
      ✓ Request-Mail-Alerts (285ms)
      ✓ Request-Mail-Internal (283ms)
      ✓ Request-Monitor-Internal (292ms)
      ✓ Request-Request-Remote (282ms)
      ✓ Request-Request-Internal (296ms)
      ✓ Request-Syscheck-Syscheck (288ms)
      ✓ Request-Syscheck-Rootcheck (280ms)
      ✓ Request-Syscheck-Internal (285ms)
      ✓ Request-Wmodules-Wmodules (285ms)


  24 passing (7s)

# mocha test/test_manager.js --grep component


  Manager
    GET/manager/config/:component/:configuration
      ✓ Request-Agentless-Agentless (311ms)
      ✓ Request-Analysis-Global (330ms)
      ✓ Request-Analysis-Active-response (428ms)
      ✓ Request-Analysis-Alerts (324ms)
      ✓ Request-Analysis-Command (311ms)
      ✓ Request-Analysis-Internal (563ms)
      ✓ Request-Auth-Auth (319ms)
      ✓ Request-Com-Active-response (305ms)
      ✓ Request-Com-Internal (312ms)
      ✓ Request-Csyslog-Csyslog (306ms)
      ✓ Request-Integrator-Integration (308ms)
      ✓ Request-Logcollector-Localfile (359ms)
      ✓ Request-Logcollector-Socket (297ms)
      ✓ Request-Logcollector-Internal (289ms)
      ✓ Request-Mail-Global (286ms)
      ✓ Request-Mail-Alerts (293ms)
      ✓ Request-Mail-Internal (293ms)
      ✓ Request-Monitor-Internal (283ms)
      ✓ Request-Request-Remote (292ms)
      ✓ Request-Request-Internal (299ms)
      ✓ Request-Syscheck-Syscheck (280ms)
      ✓ Request-Syscheck-Rootcheck (285ms)
      ✓ Request-Syscheck-Internal (306ms)
      ✓ Request-Wmodules-Wmodules (288ms)


  24 passing (8s)

# mocha test/test_agents.js --grep component


  Agents
    GET/agents/:agent/config/:component/:configuration
      ✓ Request-Agent-Client (331ms)
      ✓ Request-Agent-Buffer (445ms)
      ✓ Request-Agent-Labels (631ms)
      ✓ Request-Agent-Internal (458ms)
      ✓ Request-Agentless-Agentless (365ms)
      ✓ Request-Analysis-Global (313ms)
      ✓ Request-Analysis-Active-response (414ms)
      ✓ Request-Analysis-Alerts (317ms)
      ✓ Request-Analysis-Command (307ms)
      ✓ Request-Analysis-Internal (351ms)
      ✓ Request-Auth-Auth (315ms)
      ✓ Request-Com-Active-response (311ms)
      ✓ Request-Com-Internal (320ms)
      ✓ Request-Csyslog-Csyslog (317ms)
      ✓ Request-Integrator-Integration (316ms)
      ✓ Request-Logcollector-Localfile (356ms)
      ✓ Request-Logcollector-Socket (394ms)
      ✓ Request-Logcollector-Internal (371ms)
      ✓ Request-Mail-Global (371ms)
      ✓ Request-Mail-Alerts (355ms)
      ✓ Request-Mail-Internal (402ms)
      ✓ Request-Monitor-Internal (381ms)
      ✓ Request-Request-Remote (382ms)
      ✓ Request-Request-Internal (330ms)
      ✓ Request-Syscheck-Syscheck (413ms)
      ✓ Request-Syscheck-Rootcheck (354ms)
      ✓ Request-Syscheck-Internal (314ms)
      ✓ Request-Wmodules-Wmodules (354ms)


  28 passing (10s)
```

Best regards,
Marta

### CODE REVIEW

File: controllers/cluster.js [hunk 1]
```diff
@@ -727,5 +727,36 @@ router.get('/:node_id/configuration/validation', cache(), function(req, res) {
     execute.exec(python_bin, [wazuh_control], data_request, function (data) { res_h.send(req, res, data); });
 })
 
+/**
+ * @api {get} /cluster/:node_id/config/:component/:configuration Get active configuration in node node_id
+ * @apiName GetClusterNodeActiveConfiguration
+ * @apiGroup Configuration
+ *
+ * @apiParam {String="agent","agentless","analysis","auth","com","csyslog","integrator","logcollector","mail","monitor","request","syscheck","wmodules"} [component] Indicates the wazuh component to check.
```
Comment (crd1985) [inline controllers/cluster.js]: SCA module is missed in this list. We should include it and check if we can get that configuration on demand via socket.

File: controllers/cluster.js [hunk 2]
```diff
@@ -727,5 +727,36 @@ router.get('/:node_id/configuration/validation', cache(), function(req, res) {
...
+ * @apiDescription Returns the requested configuration in JSON format.
+ *
+ * @apiExample {curl} Example usage:
+ *     curl -u foo:bar -k -X GET "https://127.0.0.1:55000/cluster/:node_id/config/logcollector/internal?pretty"
+ *
+ */
+router.get('/:node_id/config/:component/:configuration', cache(), function(req, res) {
+    logger.debug(req.connection.remoteAddress + " GET /cluster/:node_id/config/:component/:configuration");
+
+    req.apicacheGroup = "manager";
```
Comment (crd1985) [inline controllers/cluster.js]:
```suggestion
    req.apicacheGroup = "cluster";
```

File: test/test_cluster.js [hunk 3]
```diff
@@ -1742,4 +1742,493 @@ describe('Cluster', function () {
...
+                res.body.data.wmodules[4].should.have.properties(['database']);
+                res.body.data.wmodules[5].should.have.properties(['wazuh_download']);
+
+
+                res.body.error.should.equal(0);
+                done();
+            });
+        });
+
+
```
Comment (crd1985) [inline test/test_cluster.js]: I miss some SCA module tests.

File: test/test_manager.js [hunk 4]
```diff
@@ -1268,5 +1268,492 @@ describe('Manager', function() {
...
+                res.body.data.wmodules[0].should.have.properties(['open-scap']);
+                res.body.data.wmodules[1].should.have.properties(['syscollector']);
+                res.body.data.wmodules[2].should.have.properties(['vulnerability-detector']);
+                res.body.data.wmodules[3].should.have.properties(['cis-cat']);
+                res.body.data.wmodules[4].should.have.properties(['database']);
+                res.body.data.wmodules[5].should.have.properties(['wazuh_download']);
+
+                res.body.error.should.equal(0);
+                done();
+            });
```
Comment (crd1985) [inline test/test_manager.js]: I miss some SCA tests.

File: controllers/cluster.js (hunk 1 above)
Comment (mgmacias95) [reply controllers/cluster.js]: Hello @crd1985,

SCA is already included inside `wmodules`  component and `wmodules` configuration:
```javascript
# curl -u foo:bar "localhost:55000/agents/000/config/wmodules/wmodules?pretty"
{
   "error": 0,
   "data": {
      "wmodules": [
         {
            "sca": {
               "enabled": "yes",
               "scan_on_start": "yes",
               "skip_nfs": "yes",
               "interval": 43200,
               "policies": [
                  "cis_debian_linux_rcl.yml",
                  "system_audit_rcl.yml",
                  "system_audit_ssh.yml"
               ]
            }
         }
      ]
   }
}
```

File: test/test_cluster.js (hunk 3 above)
Comment (mgmacias95) [reply test/test_cluster.js]: Fixed in https://github.com/wazuh/wazuh-api/pull/362/commits/f9d53bf3ab682d8b8ae30538f21b65990fe96c42

File: test/test_manager.js (hunk 4 above)
Comment (mgmacias95) [reply test/test_manager.js]: Fixed in https://github.com/wazuh/wazuh-api/pull/362/commits/f9d53bf3ab682d8b8ae30538f21b65990fe96c42

**Pull request merged**